### ISSUE TITLE
Add support for isolated sub-blocks

### ISSUE THREAD

**Issue by riking** [Rybadour/Foreman#29]
The planner's "Automatically create missing nodes" feature assumes all products are available globally. This is not always true - for example, it is a common feature of blue circuit builds to include local green circuit production, or for copper wire for red circuits to be isolated from copper wire for green circuits.

Allow the creation of isolated sub-graphs where the outputs and inputs are defined by the user, and intermediate nodes within are not shared with the main graph or other sub-graphs.

This style of graph is possible to create manually, but the auto-layout will always destroy this kind of graph.

**Comment by Rybadour** [comment]
Good point! I think I'd want the feature to be define as the "automatically create missing nodes" button to not connect an input to another node if it already has an input. That way you'd be able to manually create a green circuit with it's own copper wire and have it remain unconnected to the red circuit.

Also it should be much easier to disconnect nodes. At the time of writing I have no idea how to do that.

**Comment by riking** [comment]
Either right clicking the line or right clicking the input / output box, which could provide a way to select the connection to delete (by flow quantity? by type of input provider? by highlighting lines?)

The "sub-graph" phrasing is mostly for layout hints.
